src/cycleclock.h

```diff
@@ -218,6 +218,14 @@ inline BENCHMARK_ALWAYS_INLINE int64_t Now() {
   uint64_t pcycle;
   asm volatile("%0 = C15:14" : "=r"(pcycle));
   return static_cast<double>(pcycle);
+#elif defined(__alpha__)
+  // The PCC (process cycle counter) register is an unsigned 32-bit integer,
+  // which means it overflows every ~4sec depending on CPU frequency.  This
+  // tick count will only be reliable for short intervals.  This implementation
+  // is the same as what fftw uses.
+  uint64_t cc;
```

Comment:
two thoughts: if we know it's 32-bits why do we read into a 64-bit and mask?

also if it's only 4 seconds i'm not sure it's going to be very useful. is there no alternative?

Reply:
Does `gettimeofday()` (or an alternative) work on that arch? There's precedent for using that.

Reply:
> two thoughts: if we know it's 32-bits why do we read into a 64-bit and mask?
> 
> also if it's only 4 seconds i'm not sure it's going to be very useful. is there no alternative?

Hi, I helped work on this.  It's read into a 64-bit because the register is 64-bits wide, the lower 32 bits are the actual counter, while the high 32 bits are "operating system dependent".  Linux does not do anything with them today.

> Does `gettimeofday()` (or an alternative) work on that arch? There's precedent for using that.

It works, the problem is that clock precision is too low.  You can see this for example in the python bindings:

```
In [2]: time.clock_getres(time.CLOCK_MONOTONIC)
Out[2]: 0.000976563
```

This has been the cause of many a PR of mine, as on alpha two calls to `gettimeofday()` less than ~1ms apart will return the same timestamp.

That said, the `rpcc` technique seems to sufficient to pass the test suite.  I could include both options and make it configurable at compile-time perhaps?

Reply:
I don't think it makes sense to make this configurable.
It should just work right (by the definition of "right",
that is expected by the users of that architecture),
or not at all.

Reply:
> I don't think it makes sense to make this configurable. It should just work right (by the definition of "right", that is expected by the users of that architecture), or not at all.

Then `rpcc` is the way to go.  There isn't any method of measuring cycle counts for arbitrary periods on the architecture, so this is the best compromise.

Reply:
the test suite is optimised to run quickly so that's not surprising.

i don't know if we have real world users that have per iteration timings of >4s but it wouldn't surprise me if we did.  i think until there's a 64-bit high precision timer on arch, gettimeofday is going to be the better option.

Reply:
s/per-iteration/per-repetition/, which is much worse